pkg/process/runner/collector_api_test.go

```diff
@@ -424,7 +424,10 @@ func TestMultipleAPIKeys(t *testing.T) {
 
 	apiKeys := []string{"apiKeyI", "apiKeyII", "apiKeyIII"}
 
-	runCollectorTestWithAPIKeys(t, check, &endpointConfig{}, apiKeys, configmock.New(t), func(_ *CheckRunner, ep *mockEndpoint) {
+	config := configmock.New(t)
+	config.SetWithoutSource("forwarder_max_concurrent_requests", 1)
```

Comment:
The PR description already explains why this is necessary, but would you mind leaving a code comment as well so we can trace this more easily in the future?